client/web/compose/src/mixins/conditionalFields.js

```diff
@@ -0,0 +1,54 @@
+// vue mixin for conditional fields
+
+export default {
+  data () {
+    return {
+      fieldConditions: {},
+      evaluating: false,
+    }
+  },
+
+  methods: {
+    evaluateExpressions () {
+      if (!this.block.options.fieldConditions) return
+      this.evaluating = true
+      const { expressions, variables } = this.prepareFieldConditionsData()
+      return this.$SystemAPI
+        .expressionEvaluate({ variables, expressions })
+        .then(res => {
+          this.fieldConditions = res
+          this.evaluating = false
+          return res
+        }).catch(err => {
+          this.evaluating = false
+          return err
+        })
+    },
+
+    prepareFieldConditionsData () {
+      const expressions = {}
+      const variables = {
+        record: {
```

Comment:
What about system fields, make sure you include them aswell. 
In normal filters they can be accessed via the record.recordID for example.